storage3/_async/client.py

```diff
@@ -10,13 +16,42 @@
 class AsyncStorageClient(AsyncStorageBucketAPI):
     """Manage storage buckets and files."""
 
-    def __init__(self, url: str, headers: dict[str, str]) -> None:
-        super().__init__(
-            url,
-            {"User-Agent": f"supabase-py/storage3 v{__version__}", **headers},
-            AsyncClient(),
+    def __init__(
+        self,
+        url: str,
+        key: str,
+        headers: dict[str, str] = None,
+        timeout: Union[int, float, Timeout] = 5,
+    ) -> None:
+        headers = {
+            "User-Agent": f"supabase-py/storage3 v{__version__}",
+            **self._get_auth_headers(key),
+            **(headers or {}),
+        }
+        self.session = self.create_session(url, headers, timeout)
+        super().__init__(self.session)
+
+    def create_session(
+        self,
+        base_url: str,
+        headers: dict[str, str],
+        timeout: Union[int, float, Timeout],
+    ) -> AsyncClient:
+        return AsyncClient(
```

Comment:
Personally, I preferred passing the headers in each request explicitly than setting it on the client like this, because of our bad experiences with the headers and params being re-used in postgrest-py 😛

Reply:
Although I don't think it matters too much here